### DISCUSSION TITLE
Standalone build assetPrefix conditional config

### DISCUSSION THREAD

**Discussion by impaler** [vercel/next.js#47553] [Help] [answered]
### Summary

Trying to set `assetPrefix` based on a `process.env` value after a production build, does not seem possible.

Reading, does not mention this value is static and cannot change after the build:

https://nextjs.org/docs/api-reference/next.config.js/cdn-support-with-asset-prefix

It has the example of a conditional use of `process.env`

```javascript
const isProd = process.env.NODE_ENV === 'production'

module.exports = {
  // Use the CDN in production and localhost for development.
  assetPrefix: isProd ? 'https://cdn.mydomain.com' : undefined,
}
```

This led me to believe that it's possible to change this based on another process env, say for instance a custom deployment environment:

```javascript
module.exports = {
  // eg change via env config on when the server starts
  assetPrefix: process.env.CDN_PATH,
}
```

Can't get this to work in the production build as the `assetPrefix` is used in the `.next` output for various `required-server-files.json` files and generated html statically.

Anyone know how to set the `assetPrefix` dynamically, from `process.env` I am guessing its just not possible from what I can see.

**Comment by andorthehood** [comment, marked as answer]
If you opt for 'standalone' output when building, your next.config file will be serialised into ./next/standalone/server.js. As of now, a dynamic assetPrefix would be only achievable by using a custom server.

**Reply by Mihir-Karbelkar** [reply]
Can you point out to a resource for a custom server? Thanks

**Comment by moparlakci** [comment]
Just do a little string replace on the generated server.js file.

**Comment by isbasex** [comment]
Next.js official needs to support this feature as soon as possible; it is very necessary.

**Reply by bytemuck** [reply]
I cannot agree more. It really is.
